### PR TITLE
Feature/api improvements

### PR DESCRIPTION
**Pull request by jorisquist**
Improved the api by giving more control over the robot for testing. The functions to do this are all also moved to the UUV class.

Internally the pybullet is also now only imported in the Pybullet API class which is a wrapper that gives us slightly more control and adds type checking.

### CODE REVIEW

File: Main.py [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
-def move_camera_target(target):
-    camera_info = p.getDebugVisualizerCamera()
-
-    p.resetDebugVisualizerCamera(
-        cameraDistance=camera_info[10],
-        cameraYaw=camera_info[8],
-        cameraPitch=camera_info[9],
-        cameraTargetPosition=target
-    )
+# def move_camera_target(target):
```
Comment (gijskoning) [inline Main.py:10]: Should this be removed?

Comment (jorisquist) [reply Main.py:10]: yes

File: lobster_simulator/robot/Motor.py [hunk 2]
```diff
@@ -1,58 +1,72 @@
...
         else:
-            self.rpm += sign * self.max_rpm_change_per_second * dt / 1000000
+            self._rpm += sign * self._max_rpm_change_per_second * dt / 1000000
 
     def apply_thrust(self):
-        p.applyExternalForce(objectUniqueId=self.pybullet_id, linkIndex=-1,
-                             forceObj=self.direction * self.rpm_to_thrust(self.rpm),
-                             posObj=self.position,
-                             flags=p.LINK_FRAME)
+        # p.applyExternalForce(objectUniqueId=self.pybullet_id, linkIndex=-1,
```
Comment (gijskoning) [inline lobster_simulator/robot/Motor.py:64]: Should this be commented?

File: lobster_simulator/robot/Motor.py [hunk 3]
```diff
@@ -1,58 +1,72 @@
...
+        self._desired_rpm = 0
 
     @staticmethod
     def new_T200(pybullet_id, name, position, direction):
-        rpm_to_thrust = lambda rpm: 3.92 / 1000 * rpm + 3.9 / 10000000 * rpm * rpm + 7.55 / 10000000000 * rpm * rpm * rpm
-        thrust_to_rpm = lambda x: -172.185 - 5.05393 * pow(261.54 * math.sqrt(3.84767 * math.pow(10, 8) * math.pow(x, 2) + 5.13478 * math.pow(10, 8) * x + 4.48941 * math.pow(10,9)) - 5.13023 * math.pow(10, 6) * x - 3.42319 * math.pow(10, 6), (1 / 3)) + 336577. / math.pow(261.54 * math.sqrt(3.84767 * pow(10, 8) * math.pow(x, 2) + 5.13478 * math.pow(10, 8) * x + 4.48941 * math.pow(10,9)) - 5.13023 * math.pow(10, 6) * x - 3.42319 * math.pow(10, 6), (1 / 3))
-
-        return Motor(pybullet_id, name, position, direction, rpm_to_thrust, thrust_to_rpm, max_rpm_change_per_second=40000, max_rpm=4000)
+        rpm_to_thrust = lambda \
+                rpm: 3.92 / 1000 * rpm + 3.9 / 10000000 * rpm * rpm + 7.55 / 10000000000 * rpm * rpm * rpm
```
Comment (gijskoning) [inline lobster_simulator/robot/Motor.py:27]: Could you document this function a bit? Maybe reference the graph of the t200

File: lobster_simulator/robot/Motor.py (hunk 2 above)
Comment (jorisquist) [reply lobster_simulator/robot/Motor.py]: I removed it

File: lobster_simulator/robot/Motor.py (hunk 3 above)
Comment (jorisquist) [reply lobster_simulator/robot/Motor.py:31]: Yes I did this now

**Pull request merged**